rosbag2_test_common/include/rosbag2_test_common/subscription_manager.hpp

```diff
@@ -40,13 +40,11 @@ class SubscriptionManager
   }
 
   template<typename MessageT>
-  void add_subscription(const std::string & topic_name, size_t expected_number_of_messages)
+  void add_subscription(
+    const std::string & topic_name,
+    size_t expected_number_of_messages,
+    const rclcpp::QoS & qos = rclcpp::QoS{rclcpp::KeepAll()})
```

Comment:
Does systems default not make sense here? or does `KeepAll()` represent all values you removed in this particular change?

Reply:
This has come up a few times - "SystemDefault" has undefined behavior and shouldn't be used. It's not the same as "Default" - which we do define as `rosbag2_transport::Rosbag2QoS{}` which is what should be used here

Reply:
Sorry, didn't read through the rest of the reviews here on this ticket. If the default shouldn't be used, can we escalade this to `rclcpp` where it's defined? Or does it really only have UB in the context of rosbags?

Reply:
"SYSTEM_DEFAULT" in the rmw API means "let the rmw implementation use its own default values", theoretically the DDS spec defines defaults but the RMW API does not define or enforce that they are consistent across implementations. `rmw_qos_profile_system_default` is a different value than `rmw_qos_profile_default`. We are wrapping `rmw_qos_profile_default` in `Rosbag2QoS` with a default constructor and a few other rosbag2-specific utilities.